primo/data_parser/well_data.py

```diff
@@ -866,10 +874,41 @@ def _process_input_data(self):
 
     def _append_fed_dac_data(self):
         """Appends federal DAC data"""
-        census_year = self.config.census_year
-        # TODO:
-        # Append Tract ID/GEOID, population density, Total population, land area,
-        # federal DAC score to the DataFrame
+        if len(self.data) == 0:
+            # Nothing to do if no data
+            return
+
+        state_code = identify_state(self)
+        census_tracts = get_state_census_tracts(state_code, DAC_TRACT_YEAR)
+        gdf = get_data_as_geodataframe(self)
+
+        # Spatial join to identify tract id associated with every well
+        # Per 2010 data
+        gdf = gdf.sjoin(census_tracts, how="left", predicate="within")
+        self.data["Census Tract ID [2010]"] = pd.to_numeric(gdf["GEOID10"])
+
+        cejst_data = get_cejst_data()
+        joined_data = self.data.merge(
+            cejst_data,
+            left_on="Census Tract ID [2010]",
+            right_on="Census tract 2010 ID",
+            how="left",
+        )
+        self.data["Percent area disadvantaged"] = joined_data[
+            "Percentage of tract that is disadvantaged by area"
+        ]
+
+        self.data["Percent pop disadvantaged"] = joined_data[
+            "Percentage of tract that is disadvantaged by area"
+        ]
+
+        self.data["Disadvantaged Score"] = (
+            self.data["Percent area disadvantaged"]
+            + self.data["Percent pop disadvantaged"]
+        ) / 2
+
+        # Assume disadvantaged score of 0 for rows where value was not found
+        self.data["Disadvantaged Score"] = self.data["Disadvantaged Score"].fillna(0)
```

Comment:
We need to flag wells for which the data is not available. So, please replace the above line with 
`self.fill_incomplete_data(self._col_names.fed_dac, 0, "fed_dac_flag")`. This will create a new column called `fed_dac_flag` if there are any wells for which the data is not available.

Reply:
done